File: maintenance_scripts/fix_auto_delegate.mjs
/**
 * FIX AUTO-DELEGATE: Create company record for CEO user
 * Issue: GET /companies?ceo_id=... returns 400 → No metrics → Auto-Delegate không trigger
 */

import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = 'https://dqddxowyikefqcdiioyh.supabase.co';
const SUPABASE_ANON_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRxZGR4b3d5aWtlZnFjZGlpb3loIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjE3OTcxMzYsImV4cCI6MjA3NzM3MzEzNn0.okmsG2R248fxOHUEFFl5OBuCtjtCIlO9q9yVSyCV25Y';
const CEO_USER_ID = '944f7536-6c9a-4bea-99fc-f1c984fef2ef';

const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

async function fixCompanyRecord() {
  console.log('=' .repeat(60));
  console.log('🚀 FIX AUTO-DELEGATE - Create Company Record');
  console.log('='.repeat(60));
  console.log();

  try {
    // Check if CEO already has company
    console.log(`🔍 Checking for existing company (owner_id=${CEO_USER_ID.slice(0, 8)}...)`);
    
    const { data: existingCompany, error: checkError } = await supabase
      .from('companies')
      .select('id, name, owner_id')
      .eq('owner_id', CEO_USER_ID)
      .maybeSingle();

    if (checkError && checkError.code !== 'PGRST116') {
      console.error('❌ Error checking company:', checkError);
      return false;
    }

    if (existingCompany) {
      console.log(`✅ Company already exists: ${existingCompany.name} (id=${existingCompany.id})`);
      console.log('✅ Auto-Delegate should work now!');
      return true;
    }

    // Create new company
    console.log('📝 Creating new company for CEO...');
    
    const { data: newCompany, error: insertError } = await supabase
      .from('companies')
      .insert({
        name: 'SABO Tech Company',
        owner_id: CEO_USER_ID,
        created_at: new Date().toISOString(),
        updated_at: new Date().toISOString()
      })
      .select()
      .single();

    if (insertError) {
      console.error('❌ Error creating company:', insertError);
      return false;
    }

    console.log('✅ Company created successfully!');
    console.log(`   - ID: ${newCompany.id}`);
    console.log(`   - Name: ${newCompany.name}`);
    console.log(`   - CEO ID: ${CEO_USER_ID}`);

    // Verify
    const { data: verifyCompany } = await supabase
      .from('companies')
      .select(`
        id,
        name,
        owner_id
      `)
      .eq('owner_id', CEO_USER_ID)
      .single();

    if (verifyCompany) {
      console.log('\n📊 VERIFICATION:');
      console.log(`   - Company ID: ${verifyCompany.id}`);
      console.log(`   - Company Name: ${verifyCompany.name}`);
      console.log('\n🎉 Auto-Delegate is now ready to use!');
      console.log("   → Refresh browser and try: 'Giao 10 việc cấp bách cho manager'");
    }

    return true;

  } catch (error) {
    console.error('❌ Unexpected error:', error);
    return false;
  }
}

// Run the fix
fixCompanyRecord().then(success => {
  console.log();
  console.log('='.repeat(60));
  if (success) {
    console.log('✅ DONE! Auto-Delegate should work now!');
  } else {
    console.log('❌ FAILED! Check errors above.');
  }
  console.log('='.repeat(60));
  process.exit(success ? 0 : 1);
});

File: maintenance_scripts/fix_auto_delegate.py
#!/usr/bin/env python3
"""
FIX AUTO-DELEGATE: Create company record for CEO user
Issue: GET /companies?ceo_id=... returns 400 → No metrics → Auto-Delegate không trigger
"""

import os
import psycopg2
from datetime import datetime

# Database connection from .env
DATABASE_URL = "postgresql://postgres.dqddxowyikefqcdiioyh:Acookingoil123@aws-1-ap-southeast-2.pooler.supabase.com:6543/postgres"
CEO_USER_ID = "944f7536-6c9a-4bea-99fc-f1c984fef2ef"

def fix_company_record():
    """Create company record for CEO if not exists"""
    try:
        # Connect to database
        print("🔌 Connecting to Supabase...")
        conn = psycopg2.connect(DATABASE_URL)
        cursor = conn.cursor()
        
        # Check if CEO already has company
        print(f"🔍 Checking for existing company (owner_id={CEO_USER_ID[:8]}...)")
        cursor.execute("""
            SELECT id, name, owner_id 
            FROM companies 
            WHERE owner_id = %s
        """, (CEO_USER_ID,))
        
        existing = cursor.fetchone()
        
        if existing:
            print(f"✅ Company already exists: {existing[1]} (id={existing[0]})")
            print("✅ Auto-Delegate should work now!")
            return True
        
        # Create new company
        print("📝 Creating new company for CEO...")
        cursor.execute("""
            INSERT INTO companies (name, owner_id, created_at, updated_at)
            VALUES (%s, %s, %s, %s)
            RETURNING id, name
        """, (
            'SABO Tech Company',
            CEO_USER_ID,
            datetime.now(),
            datetime.now()
        ))
        
        new_company = cursor.fetchone()
        conn.commit()
        
        print(f"✅ Company created successfully!")
        print(f"   - ID: {new_company[0]}")
        print(f"   - Name: {new_company[1]}")
        print(f"   - CEO ID: {CEO_USER_ID}")
        
        # Verify
        cursor.execute("""
            SELECT 
                c.id as company_id,
                c.name,
                c.owner_id,
                u.email as ceo_email
            FROM companies c
            LEFT JOIN auth.users u ON u.id = c.owner_id
            WHERE c.owner_id = %s
        """, (CEO_USER_ID,))
        
        verify = cursor.fetchone()
        if verify:
            print("\n📊 VERIFICATION:")
            print(f"   - Company ID: {verify[0]}")
            print(f"   - Company Name: {verify[1]}")
            print(f"   - CEO Email: {verify[3] or 'N/A'}")
            print("\n🎉 Auto-Delegate is now ready to use!")
            print("   → Refresh browser and try: 'Giao 10 việc cấp bách cho manager'")
        
        cursor.close()
        conn.close()
        return True
        
    except Exception as e:
        print(f"❌ Error: {e}")
        return False

if __name__ == "__main__":
    print("=" * 60)
    print("🚀 FIX AUTO-DELEGATE - Create Company Record")
    print("=" * 60)
    print()
    
    success = fix_company_record()
    
    print()
    print("=" * 60)
    if success:
        print("✅ DONE! Auto-Delegate should work now!")
    else:
        print("❌ FAILED! Check errors above.")
    print("=" * 60)
